### PR TITLE
fix(image): fix problem with open jar files

### LINKED ISSUE #3514: Unable to open JAR files
## Description
Java scanning may lead to the following error.

```
failed to analyze file: failed to analyze usr/lib/jvm/java-1.8-openjdk/lib/tools.jar: unable to open usr/lib/jvm/java-1.8-openjdk/lib/tools.jar: failed to open: unable to read the file: stream error: stream ID 9; PROTOCOL_ERROR; received from peer
```

Currently, we're investigating this issue. As a temporary mitigation, you may be able to avoid this issue by downloading the Java DB in advance. `--download-java-db-only` is available in v0.37.0+.

```
$ trivy image --download-java-db-only
2023-02-01T16:57:04.322+0900    INFO    Downloading the Java DB...
$ trivy image [YOUR_JAVA_IMAGE]
```

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
fixes for following error:
`PROTOCOL_ERROR; received from peer`  - Increase timeout for getting remote image to fix 
`unexpected EOF`  - Save layers smaller than 50MB to files and use this to parse files. 

## Related issues
- Close #3514

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/artifact/image/image.go [hunk 1]
```diff
@@ -335,33 +335,38 @@ func (a Artifact) diffIDs(configFile *v1.ConfigFile) []string {
...
+		return "", 0, nil, xerrors.Errorf("failed to get the layer size (%s): %w", diffID, err)
+	}
+
 	rc, err := layer.Uncompressed()
 	if err != nil {
-		return "", nil, xerrors.Errorf("failed to get the layer content (%s): %w", diffID, err)
+		return "", 0, nil, xerrors.Errorf("failed to get the layer content (%s): %w", diffID, err)
 	}
-	return digest, rc, nil
+	return digest, size, rc, nil
```
Comment (knqyf263) [inline pkg/fanal/artifact/image/image.go:369]: I think we can return the cached layer here. Also, we should always cache layers when `--slow` is passed.

Comment (knqyf263) [reply pkg/fanal/artifact/image/image.go:369]: In addition, cachedFile is no longer needed.

Comment (DmitriyLewen) [reply pkg/fanal/artifact/image/image.go]: Done

**Comment by DmitriyLewen** [comment]
These fixes have been added to the following PRs:
- #3582
- #3640

**Pull request closed without merging**